.github/workflows/benchmarking.yml

```diff
@@ -94,10 +94,31 @@ jobs:
           name: DerivedData-Xcode
       - run: npm install -g saucectl@0.186.0
       - name: Run Benchmarks in SauceLab
+        id: run-benchmarks-in-sauce-lab
         env:
           SAUCE_USERNAME: ${{ secrets.SAUCE_USERNAME }}
           SAUCE_ACCESS_KEY: ${{ secrets.SAUCE_ACCESS_KEY }}
-        run: saucectl run --select-suite "${{matrix.suite}}" --config .sauce/benchmarking-config.yml --tags benchmark
+        run: saucectl run --select-suite "${{matrix.suite}}" --config .sauce/benchmarking-config.yml --tags benchmark --retries 3
+      - name: Run Benchmarks in SauceLab - Retry 1
+        id: run-benchmarks-in-sauce-lab-retry-1
+        if: ${{ steps.run-benchmarks-in-sauce-lab.outcome == 'failure' }}
+        env:
+          SAUCE_USERNAME: ${{ secrets.SAUCE_USERNAME }}
+          SAUCE_ACCESS_KEY: ${{ secrets.SAUCE_ACCESS_KEY }}
+        run: saucectl run --select-suite "${{matrix.suite}}" --config .sauce/benchmarking-config.yml --tags benchmark --retries 3
```

Comment:
`m`: I would only retry if we get the specific 500 error. Otherwise, we also retry when UI tests are flaky, which will lead to flaky tests staying unnoticed for some time and will increase CI time.

Reply:
I agree, but I don't see a reliable way of detecting the error cause other than trying to extract information from the error log - which also has no clear indicator

Reply:
In case we want to go for that approach, I addded additional workflow jobs that failed to #4689 and grouped them by their log message

Reply:
What about always retrying 3 times unless there is a failing test? We should be able get the info if a test failed in Saucelabs. If it did we don't retry.

Reply:
That could work, I'll look into this, but we should also retry when tests time out

Reply:
Yes, please.

Reply:
But you can do timeouts later. Step by step.

Reply:
https://github.com/getsentry/sentry-cocoa/actions/runs/12710298356/job/35431329854?pr=4694

I added a simulated failure.

Using `continue-on-error` following jobs are executed even when the workflow step fails. I had to extract the test id from the build log to get the test result from `saucectl`.

Not happy with this approach, even though it works, extraction of an ID from the log is unstable.